src/core/lib/iomgr/error_internal.h

```diff
@@ -48,12 +48,12 @@ struct grpc_linked_error {
 
 struct grpc_error {
   gpr_refcount refs;
+  gpr_atm error_string;
```

Comment:
I chose to do 1, I agree that that is cleaner, and seemed worth the bigger modification of the code. I also took this opportunity to be a bit less terse with my comments :)